weaviate/gql/get.py

```diff
@@ -111,12 +123,14 @@ def __init__(
             raise TypeError(
                 "properties must be of type str, " f"list of str or None but was {type(properties)}"
             )
-        for prop in properties:
-            if not isinstance(prop, str):
-                raise TypeError("All the `properties` must be of type `str`!")
+
+        self._properties: List[Union[str, Reference]] = []
+        for _, prop in enumerate(properties):
```

Comment:
why not?
```python
for prop in properties:
```

Reply:
no good reason, probably used the index at some point :)